### ISSUE TITLE
Move image "ALT" indicator away from bottom-left (perhaps to bottom-right)

### ISSUE THREAD

**Issue by jsstaedtler** [sk22/megalodon#556]
A request for a minor improvement:

When browsing a feed in Mastodon, images with (or without) alt text will exhibit a little black "ALT" icon (or red "!" icon) in the bottom-left corner of each image.

The trouble is, if the image contains a paragraph or more of text (like a screenshot of a social media post or web page), that text will almost always extend to the bottom-left corner of the image.  And it will always be obscured by Mastodon's ALT icon.  Of course, a workaround is to tap the image to open it full-screen, but it's a mild annoyance just to read that one dang word in the bottom-left.

Could the icon be moved elsewhere?  I propose the bottom-right corner, where there is less likely to be text in an image.

I recognize one caveat: that's where the "GIF" indicator normally resides.  Perhaps that could move to the bottom-left instead (since animated GIFs are less likely to contain a block of text)?

**Comment by jsstaedtler** [comment]
Oh, and I was only thinking of text written left-to-right, which is a bit of a bad assumption on my part.  Making this change would effectively hand the problem over to people who write right-to-left.  There's obviously no perfect solution, the indicator is going to cover up *something*... is it too extreme to base it on the user's system language setting?  That could give a high probability of putting the icon in the less-obtrusive corner.

**Comment by sk22** [comment]
good idea! not 100% sure if this change wouldn't be a bit too disruptive to users who're already used to it being on the left (megalodon/mastodon isn't the only app that does it like this), or if it might just look out of place. i'd definitely at least try and see how it looks. but then: where would i move the GIF indicator (which is currently on the bottom right)? (no need to reply to this; it's just something that i'd have to consider)

> is it too extreme to base it on the user's system language setting

not at all, that's what "start" and "end" alignments are for :D actually, anything "left" and "right" is generally discouraged to use in android layouts. when you look at megalodon (or mastodon) in left-to-right mode, you already see the alt text being on the opposite side of the screen!

**Comment by KevinWh0** [comment]
Maybe add it as a setting to select what corner to place it in? That way both right to left and left to right readers win.

**Comment by d4f5409d** [comment]
Oh no

**Comment by d4f5409d** [comment]
What are you talking about? I checked and moving to the right is absolutely nonsense.![Screenshot_2023-07-21-22-32-20-76_97294d061ac208a4e8e283a3feb896c9.jpg](https://github.com/sk22/megalodon/assets/76624594/c219ecaa-797b-4535-8a94-85707a0c5e22)

**Comment by sk22** [comment]
upstream mastodon for android had a change in the material 3 redesign where they put the GIF badge on the left - next to the ALT badge, which doesn't look good imo. leaving the GIF badge on the left, it makes total sense to remove the ALT badge to the right

**Comment by sk22** [comment]
next megalodon version:

![Screenshot_20230722-011532_Megalodon](https://github.com/sk22/megalodon/assets/6217438/4baded56-579b-45d8-973c-9cedac9f01f9)

current upstream mastodon version:

![Screenshot_20230722-011547_Mastodon](https://github.com/sk22/megalodon/assets/6217438/61b8ab1c-e882-4b97-a2cd-22165532ee10)

**Comment by d4f5409d** [comment]
Why do you follow the upstream in this case? It's their mess, and yes we've got already used to it, so please don't cchange it's position. Thanks
